### PR TITLE
Create Rails App

### PR DESCRIPTION
**Pull request by Theophileaseh**
## In this project milestone, I implemented the following project requirements:

- Create a new Rails API app called 'hello-rails-back-end'.
- Initialize your project with Git.
- Make sure that your project has a Postgres database set up.
- Create a table for storing your messages and create 5 different greetings.
- Create an API endpoint that selects a random greeting from your table (you will need a controller with an action an rails route).

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Rails.application.routes.draw do
+  root 'messages#index'
+  resources :messages
+
+  namespace :api do
+    namespace :v1 do
+      resources :messages, only: %i[index]
+    end
+  end
```
Comment (TedLivist) [inline config/routes.rb:9]: - [ ] The react app is unable to fetch data from this backend due to CORS policy. You can fix this by installing the `gem 'rack-cors'` and adding a `config/initializers/cors.rb` file and allow requests from all sites. You can visit the documentation here: [rack-cors gem documentation](https://github.com/cyu/rack-cors).

  fkfe

File: app/controllers/api/v1/messages_controller.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Api::V1::MessagesController < ApplicationController
+  def index
+    @messages = Message.all
+    render json: @messages
+  end
```
Comment (TedLivist) [inline app/controllers/api/v1/messages_controller.rb]: - [ ] After fixing the issue above, you still need to configure your backend and the frontend to display a **random** greeting, not all greetings. Every time the front end is refreshed the API should return a random message and the Frontend should display it. For example, you can change this method to the code below and refactor the FE as well to handle it.

  ```
  def index
    message = Message.all.sample
    render json: { message: message }
  end
  ``` 

  fjfk

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Theophileaseh,

Good job so far!
✔️ Good documentation
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by julie-ify** [comment]
Hi @Theophileaseh,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly implement the changes requested by the previous reviewer

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by rloterh — COMMENTED** [review]
Hi Nwachan,
 
 While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
 
 ## Invalid Code Review Request
You have submitted a project, whereby Rubocop linter checks are failing on your Pull Request. Kindly ensure that **all** checks are passing and request for another code review.

**Rubocop linter check error:**
image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
 Cheers and Happy coding!👏👏👏
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _Invalid Code Review Request **does not count** into the code reviews limit._

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @Theophileaseh👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**